src/mpi_native/mpi_native.cpp

```diff
@@ -47,28 +65,777 @@ int MPI_Init(int* argc, char*** argv)
 int MPI_Comm_rank(MPI_Comm comm, int* rank)
 {
     auto logger = faabric::util::getLogger();
-    logger->debug("MPI_Comm_rank");
+    logger->debug("S - MPI_Comm_rank");
 
     *rank = executingContext.getRank();
 
     return MPI_SUCCESS;
 }
 
 int MPI_Comm_size(MPI_Comm comm, int* size)
+{
+    auto fptr = &faabric::scheduler::MpiWorld::getSizePtr;
+    callMpiFunc("MPI_Comm_size", fptr, size);
+
+    return MPI_SUCCESS;
+}
+
+int MPI_Finalize()
+{
+    auto logger = faabric::util::getLogger();
+    logger->debug("S - MPI_Finalize");
+
+    return MPI_SUCCESS;
+}
+
+int MPI_Get_version(int* version, int* subversion)
+{
+    notImplemented("MPI_Get_version");
+
+    return MPI_SUCCESS;
+}
+
+int MPI_Send(const void* buf,
+             int count,
+             MPI_Datatype datatype,
+             int dest,
+             int tag,
+             MPI_Comm comm)
+{
+    auto fptr = &faabric::scheduler::MpiWorld::send;
+    callMpiFunc(
+      fmt::format("MPI_Send {} -> {}", executingContext.getRank(), dest),
+      fptr,
+      executingContext.getRank(),
+      dest,
+      (uint8_t*)buf,
+      datatype,
+      count,
+      faabric::MPIMessage::NORMAL);
```

Comment:
I'm not convinced by this function pointer approach. It adds a lot of indirection and we're now having to modify the main MPI code to support it, yet in this case I'm not sure it actually saves any lines of code.

Without the func pointer stuff (in the case where we only need `logger` once, and `world` once):

```
faabric::util::getLogger()->debug("MPI_Send {} -> {}", executingContext.getRank(), dest);

getExecutingWorld().send(executingContext.getRank(), 
...
```

With the function pointer stuff:

```
callMpiFunc(fmt::format("MPI_Send {} -> {}", executingContext.getRank(), dest),
    &faabric::scheduler::MpiWorld::send,
    executingContext.getRank(),
    ...
```

I think the first one is much easier to follow personally, and doesn't seem that much more verbose. If you want to save more characters you could also add `using namespace faabric::util` and/ or `using namespace faabric::scheduler`.